mongodbatlas/fw_data_source_mongodbatlas_atlas_user.go

```diff
@@ -0,0 +1,254 @@
+package mongodbatlas
+
+import (
+	"context"
+	"fmt"
+	"time"
+
+	"github.com/hashicorp/terraform-plugin-framework-validators/stringvalidator"
+	"github.com/hashicorp/terraform-plugin-framework/datasource"
+	"github.com/hashicorp/terraform-plugin-framework/datasource/schema"
+	"github.com/hashicorp/terraform-plugin-framework/path"
+	"github.com/hashicorp/terraform-plugin-framework/schema/validator"
+	"github.com/hashicorp/terraform-plugin-framework/types"
+	"go.mongodb.org/atlas-sdk/v20230201002/admin"
+)
+
+const (
+	AtlasUserDataSourceName = "atlas_user"
+	errorUserRead           = "error getting atlas users(%s): %s"
+)
+
+var _ datasource.DataSource = &AtlasUserDS{}
+var _ datasource.DataSourceWithConfigure = &AtlasUserDS{}
+
+type tfAtlasUserDSModel struct {
+	ID           types.String           `tfsdk:"id"`
+	UserID       types.String           `tfsdk:"user_id"`
+	Username     types.String           `tfsdk:"username"`
+	Country      types.String           `tfsdk:"country"`
+	CreatedAt    types.String           `tfsdk:"created_at"`
+	EmailAddress types.String           `tfsdk:"email_address"`
+	FirstName    types.String           `tfsdk:"first_name"`
+	LastAuth     types.String           `tfsdk:"last_auth"`
+	LastName     types.String           `tfsdk:"last_name"`
+	MobileNumber types.String           `tfsdk:"mobile_number"`
+	TeamIDs      []string               `tfsdk:"team_ids"`
+	Links        []tfLinkModel          `tfsdk:"links"`
+	Roles        []tfAtlasUserRoleModel `tfsdk:"roles"`
+}
+
+type tfLinkModel struct {
+	Href types.String `tfsdk:"href"`
+	Rel  types.String `tfsdk:"rel"`
+}
+
+type tfAtlasUserRoleModel struct {
+	GroupID  types.String `tfsdk:"group_id"`
+	OrgID    types.String `tfsdk:"org_id"`
+	RoleName types.String `tfsdk:"role_name"`
+}
+
+func NewAtlasUserDS() datasource.DataSource {
+	return &AtlasUserDS{}
+}
+
+type AtlasUserDS struct {
+	client *MongoDBClient
+}
+
+func (d *AtlasUserDS) Metadata(ctx context.Context, req datasource.MetadataRequest, resp *datasource.MetadataResponse) {
+	resp.TypeName = fmt.Sprintf("%s_%s", req.ProviderTypeName, AtlasUserDataSourceName)
+}
+
+func (d *AtlasUserDS) Configure(ctx context.Context, req datasource.ConfigureRequest, resp *datasource.ConfigureResponse) {
+	client, err := ConfigureClient(req.ProviderData)
+	if err != nil {
+		resp.Diagnostics.AddError(errorConfigureSummary, err.Error())
+		return
+	}
+	d.client = client
+}
+
+func (d *AtlasUserDS) Schema(ctx context.Context, req datasource.SchemaRequest, resp *datasource.SchemaResponse) {
+	resp.Schema = schema.Schema{
+		Attributes: map[string]schema.Attribute{
+			"id": schema.StringAttribute{ // required by hashicorps terraform plugin testing framework
+				DeprecationMessage:  "Please use user_id id attribute instead",
```

Comment:
Can't we just set `id=user_id` for this resource?

Reply:
I see that we are already doing this. Why do we have a deprecation message?

Reply:
Relevant github issue: https://github.com/hashicorp/terraform-plugin-testing/issues/84#issuecomment-1480006432.
Essentially if no id attribute is defined in schema, acceptance tests throw an error  no "id" found in attributes. This seems to be a legacy check related to sdk v2, so if id attribute is not needed the suggestion is to define and leave as deprecated.
I can add this link in the comment to avoid confusions here.